wordpress-mobile/WordPress-Android
# [Build] Remove Gradle Enterprise from Project

Since `Gradle Enterprise (GE)` is shut down, at least temporarily (see `paaHJt-4Ap-p2`), this PR removes the existing `Gradle Enterprise` configuration as well (234a185ae7aa4a6dedf1b4aaf9a0234eedcfbd8a + 79d80cef8236a45185c55784a9a5459dee524b6a).

Also, this PR adds back the previously removed custom Gradle related `buildScan` and `buildCache` configuration (0a79ca165be9ecd106f0c4557808fa81e691f5eb + 89efb1754416bbd0db1cf59c236e29f078736ea2).

FYI: While adding back the previously removed custom Gradle related `buildScan` and `buildCache` configuration, while the previous configuration only targeted `CI`, this new configuration also enables those for `LOCAL` builds. 🤞 

PS: I also chose not to bring back the `GRADLE_ENTERPRISE_ANALYTICS_DISABLE`, `GRADLE_ENTERPRISE_ANALYTICS_USERNAME`, and `GRADLE_ENTERPRISE_ANALYTICS_HOSTNAME` configuration that was added to `buildScan` afterwards (#16489). And this was done due to the fact that I don't think this kind of obfuscation is still necessary for when working without `Gradle Enterprise`. 🤔

-----

To test:

- (`GE`): Verify `LOCAL` and `CI` builds no longer output the `A network error occurred.` error at the end of the build, related to it failing to connect to `Gradle Enterprise` since it has been shut down for day now.
- (`buildScan`): Verify `LOCAL` and `CI` builds are publishing the build scan at the end of the build, and create the corresponding build scan link.
- (`buildCache`): Verify `LOCAL` and `CI` builds are able to use the remote build cache. ⚠️

-----

The ⚠️ above on the `buildCache` functionality is due to the fact that even during my local testing, after re-configuring `buildCache`, it seems that builds are failing to connect to the remove http build cache ([url](http://10.0.2.215:5071/cache/)), see specific errors below:
  - `Could not load entry 2808becc7dc09f5e801fd94fca56cd08 from remote build cache: Connect to 10.0.2.215:5071 [/10.0.2.215] failed: Connect timed out`
  - `The remote build cache was disabled during the build due to errors.`

~~Thus, I added the `[Do Not Merge]` title prefix and the `[Status] Not Ready for Merge` label.~~

**UPDATE**: `buildCache` (and `buildScan`) got disabled for local builds, as such, this ⚠️ is no longer relevant as it was only applicable to local builds.

-----
## Regression Notes

1. Potential unintended areas of impact

    - Build scan is not working as expected or makes `LOCAL` build slower than usual (due to `uploadInBackground` misconfiguration or otherwise).
    - Build cache is not working as expected or makes builds slower than usual (due to slow download speeds or otherwise).

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
